frontend/benefit/applicant/src/components/applications/forms/application/step1/extended.ts

```diff
@@ -85,22 +91,27 @@ const useComponent = (): ExtendedComponentProps => {
   );
 
   const fields = React.useMemo((): FieldsDef => {
+    const fieldMasks: Record<Field['name'], Field['mask']> = {
+      [APPLICATION_FIELDS.COMPANY_IBAN]: {
+        format: 'FI99 9999 9999 9999 99',
+        stripVal: (val: string) => val.replace(/\s/g, ''),
+      },
+    };
+
     const fieldsdef: FieldsDef = {};
     fieldNames.forEach((name) => {
```

Comment:
This ^ returns an `array` rather than a dictionary. I tried to refactor it in a more functional fashion using `reduce` but `eslint` started complaining. Apparently, `reduce` has been intentionally discouraged!

```TypeScript

    return fieldNames.reduce<FieldsDef>(
      (prev, name) => ({
        ...prev,
        [name]: {
          name,
          label: t(`${translationsBase}.fields.${name}.label`),
          placeholder: t(`${translationsBase}.fields.${name}.placeholder`),
          mask: fieldMasks[name],
        },
      }),
      {}
    );
```

Reply:
IMHO Just disable the no-reduce rule, it's too opinionated. Eslint-auto is very strict but good and auto-scaled lint-package generally but this rule is just too strict.

Reply:
Agreed and fixed!